packages/flame/lib/src/collisions/hitboxes/hitbox.dart

```diff
@@ -0,0 +1,30 @@
+import '../../../collisions.dart';
+import '../../../extensions.dart';
+
+/// The [Hitbox] is the default building block to determine whether two objects
+/// have collided with each other. [ShapeHitbox] is the default implementation
+/// used in FCS.
+mixin Hitbox<T extends Hitbox<T>> implements GenericCollisionCallbacks<T> {
```

Comment:
This looks more like an abstract class to be implemented, not a mixin.
Conceptually, a mixin cannot be "a building block".

Reply:
Will convert it to an abstract class, didn't know you could mixin classes just like mixins (`HitboxShape` needs to use it as a mixin).

Reply:
Fixed

Reply:
:+1: